frame/sudo/src/tests.rs

```diff
@@ -0,0 +1,67 @@
+use super::*;
+use mock::{
+	Sudo, Origin, Call, Test, PrivelegedCall, new_test_ext,
+}; 
+use frame_support::{assert_ok, assert_err};
+
+#[test]
+fn new_test_ext_and_sudo_get_key_works() {
+	// Test that the enivroment setup and pallets key retrieval works as expected
+	new_test_ext(1).execute_with(|| {
+		assert_eq!(Sudo::key(),  1u64);
+	});
+}
+
+#[test]
+fn sudo_basics() {
+	new_test_ext(1).execute_with(|| {
+		// A privelleged function should work when sudo is passed the root_key as origin
+		let call = Box::new(Call::Priveleged(PrivelegedCall::privileged_function()));
+		assert_ok!(Sudo::sudo(Origin::signed(1), call));
+		// A privelleged function should not work when called with a non-root user
+		let call = Box::new(Call::Priveleged(PrivelegedCall::privileged_function()));
+		assert_err!(Sudo::sudo(Origin::signed(2), call), Error::<Test>::RequireSudo);
```

Comment:
all runtime functions should use `assert_noop` which not only confirm the error, but also confirms nothing changed in the Runtime storage. (the storage root is the same before and after)

Essentially, it checks the "verify first, write last"

Reply:
So all `assert_err` should be turned into `assert_noop`

Reply:
All ```assert_err```s have been removed. ```assert_noop``` is used when applicable.